ui-speedspacechart/src/components/utils.ts

```diff
@@ -326,26 +326,38 @@ export const interpolate = (x1: number, y1: number, x2: number, y2: number, x: n
 export const clamp = (value: number, min: number, max: number) =>
   Math.min(Math.max(value, min), max);
 
-/** Filter stops to avoid overlapping draw */
+/** Filter stops to avoid overlapping, showing the most important stops first */
 export const filterStops = (
-  stops: LayerData<string>[],
+  stops: LayerData<OperationalPoints>[],
   ratioX: number,
   width: number,
   maxPosition: number,
   minSpace = 8
 ) => {
-  let lastDisplayedPosX: number | null = null;
+  // Sort stops by decreasing weight to prioritize the heaviest ones
+  const sortedStops = stops.slice().sort((a, b) => (b.value.weight ?? 0) - (a.value.weight ?? 0));
 
-  const filteredStops = stops.filter(({ position }) => {
-    const posX = positionToPosX(position.start, maxPosition, width, ratioX);
-    if (lastDisplayedPosX !== null && Math.abs(posX - lastDisplayedPosX) < minSpace) {
-      return false;
+  const displayedStops: LayerData<OperationalPoints>[] = [];
+  const occupiedPositions: number[] = [];
```

Comment:
Maybe we can use a single array of object which have the stops and the position within.
```suggestion
const displayedStops = Array<{ stop : LayerData<OperationalPoints>, posX:number }>
```

Reply:
fixed